ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_math.py

```diff
@@ -559,3 +559,37 @@ def test_paddle_abs(
         on_device=on_device,
         x=x[0],
     )
+
+
+# cumprod
+@handle_frontend_test(
+    fn_tree="paddle.tensor.math.cumprod",
+    dtype_and_x=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("valid"),
+        safety_factor_scale="log",
+        small_abs_safety_factor=20,
+        min_num_dims=1,
+        valid_axis=True,
+        force_int_axis=True,
+    ),
```

Comment:
please also draw values for the `dtype` arguments with this helper: https://github.com/unifyai/ivy/blob/9d69a6737927afe81fcf4ba44707c8bfa2673923/ivy_tests/test_ivy/helpers/hypothesis_helpers/dtype_helpers.py#L84

Reply:
Hello @juliagsy , I added dtype argument as you mentioned above and made necessary changes in the testing function as below.
![image](https://github.com/unifyai/ivy/assets/86348193/5e66d520-011f-49d6-a721-025330fcff89)
Now, I am getting below type error with the failed tests - 
![image](https://github.com/unifyai/ivy/assets/86348193/9598cf5d-fc1f-425a-933f-9fc1b1906b46)
I tried with different supported dtypes but each time the error is same, with the respective mentioned dtype.

Reply:
the `dtype` sampling should have the parameter `full=False` because we should only sample 1 dtype instead of the full list

Reply:
Hi @juliagsy , I added the above mentioned argument to dtype, but still it's showing same error.

Reply:
@sachelsout hmm, this shouldn't happen, I'll dive into it a bit more and come back to you! Thanks!

Reply:
@juliagsy Sure, thank you so much !!

Reply:
Hey @sachelsout ! Apologies for the delayed response!
I've found the cause of issues and it's because when we try to get the native results, we convert all arguments to the backend equivalent, and pass them to the native function, but paddle by default doesn't take paddle dtype in the `dtype` argument (which is quite weird), therefore the error
Sorry for the trouble but would you mind to keep the `dtype` argument out of the test for now? We'll then figure it out internally.
Please ping me again when the related changes are done, thanks a lot!

Reply:
Hi @juliagsy , thank you for briefly explaining as why we were facing the dtype related issue. I'll make the necessary changes and ping you ! 😃